modules/aws_roles/variables.tf

```diff
@@ -0,0 +1,33 @@
+variable cluster_id {
+    description = "cluster ID"
+    type = string
+}
+
+variable rh_oidc_provider_url {
+    description = "oidc provider url"
+    type = string
+    default = "rh-oidc.s3.us-east-1.amazonaws.com"
+}
+
+variable rh_oidc_provider_thumbprint {
+    description = "Thumbprint for https://rh-oidc.s3.us-east-1.amazonaws.com"
+    type = string
+    default = "917e732d330f9a12404f73d8bea36948b929dffc"
+}
+
+variable operator_roles_properties {
+    description = "List of ROSA Operator IAM Roles"
+    type = list(object({
+        role_name = string
+        policy_name = string
+        service_accounts = list(string)
+        operator_name = string
+        operator_namespace = string
+        role_arn = string
+    }))
+    validation {
+      condition     = length(var.operator_roles_properties) == 6
```

Comment:
Not sure if this should be fixed. 4.9 clusters for instance have 5 and in the future newer versions might require more than 6.

Reply:
I didn't find a way to make it dynamic- terraform doesn't support a dynamic creation because in the planning stage it determines all the resources that should be created... If the number is not a "const" it doesn't know how much resources to create... 

I got an error when I changed it to a calculated in runtime number: 
`The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot... To work around this, use the -target argument to first apply only the resources that the count depends on`


For that PR I can add a validation about the cluster version...

Reply:
I added a cluster version validation in a separate commit